File: safedialog/src/main/java/sgtmelon/safedialog/library/MultiplyDialog.java
package sgtmelon.safedialog.library;

import android.app.Dialog;
import android.os.Bundle;

import java.util.Arrays;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import sgtmelon.safedialog.R;
import sgtmelon.safedialog.office.annot.DialogAnn;
import sgtmelon.safedialog.office.blank.DialogBlank;

public final class MultiplyDialog extends DialogBlank {

    private String[] name;
    private boolean[] init, check;

    public void setArguments(boolean[] check) {
        final Bundle bundle = new Bundle();

        bundle.putBooleanArray(DialogAnn.INIT, check.clone());
        bundle.putBooleanArray(DialogAnn.VALUE, check);

        setArguments(bundle);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final Bundle bundle = getArguments();

        if (savedInstanceState != null) {
            init = savedInstanceState.getBooleanArray(DialogAnn.INIT);
            check = savedInstanceState.getBooleanArray(DialogAnn.VALUE);
        } else if (bundle != null) {
            init = bundle.getBooleanArray(DialogAnn.INIT);
            check = bundle.getBooleanArray(DialogAnn.VALUE);
        }

        return new AlertDialog.Builder(context)
                .setTitle(title)
                .setMultiChoiceItems(name, check, (dialog, which, isChecked) -> {
                    check[which] = isChecked;
                    setEnable();
                })
                .setPositiveButton(getString(R.string.dialog_btn_accept), onPositiveClick)
                .setNegativeButton(getString(R.string.dialog_btn_cancel), (dialog, id) -> dialog.cancel())
                .setCancelable(true)
                .create();
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);

        outState.putBooleanArray(DialogAnn.INIT, init);
        outState.putBooleanArray(DialogAnn.VALUE, check);
    }

    public void setRows(String[] name) {
        this.name = name;
    }

    public boolean[] getCheck() {
        return check;
    }

    @Override
    protected void setEnable() {
        super.setEnable();

        if (Arrays.equals(init, check)) buttonPositive.setEnabled(false);
        else buttonPositive.setEnabled(true);
    }

}

File: safedialog/src/main/java/sgtmelon/safedialog/library/MultiplyDialog.kt
package sgtmelon.safedialog.library

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import sgtmelon.safedialog.R
import sgtmelon.safedialog.office.annot.DialogAnn
import sgtmelon.safedialog.office.blank.DialogBlank
import java.util.*

class MultiplyDialog : DialogBlank() {

    lateinit var name: Array<String>

    private lateinit var init: BooleanArray

    lateinit var check: BooleanArray
        private set

    fun setArguments(check: BooleanArray) {
        val bundle = Bundle()

        bundle.putBooleanArray(DialogAnn.INIT, check.clone())
        bundle.putBooleanArray(DialogAnn.VALUE, check)

        arguments = bundle
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val bundle = arguments

        init = savedInstanceState?.getBooleanArray(DialogAnn.INIT)
                ?: bundle?.getBooleanArray(DialogAnn.INIT)
                ?: BooleanArray(size = 0)

        check = savedInstanceState?.getBooleanArray(DialogAnn.VALUE)
                ?: bundle?.getBooleanArray(DialogAnn.VALUE)
                ?: BooleanArray(size = 0)

        return AlertDialog.Builder(context!!)
                .setTitle(title)
                .setMultiChoiceItems(name, check) { _, which, isChecked ->
                    check[which] = isChecked
                    setEnable()
                }
                .setPositiveButton(getString(R.string.dialog_btn_accept), onPositiveClick)
                .setNegativeButton(getString(R.string.dialog_btn_cancel)) { dialog, _ -> dialog.cancel() }
                .setCancelable(true)
                .create()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        outState.putBooleanArray(DialogAnn.INIT, init)
        outState.putBooleanArray(DialogAnn.VALUE, check)
    }

    override fun setEnable() {
        super.setEnable()
        buttonPositive.isEnabled = !Arrays.equals(init, check)
    }

}